### PR TITLE
Options processor + tests

### PR DESCRIPTION
**Pull request by shine2lay**
@jusx This is an independent Unit, the CI should be passing for this, please review this one first

### CODE REVIEW

File: lib/validators/options_processor/options.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+const consolidateResult = require('./options/lib/consolidateResults')
+const constructOupt = require('./options/lib/constructOutput')
```
[the remark this answers is not in the document]
Comment (shine2lay) [reply lib/validators/options_processor/options.js]: done

**Comment by shine2lay** [comment]
@jusx re process class: we are currently using static method, your method would require that we instantiate the class i believe, how can we go about this without instantiating the class

**Comment by jusx** [comment]
I don't mind instantiating the class. The code is the cleanest IMO.

**Comment by shine2lay** [comment]
@jusx we already had a discussion about not instantiating options processor as it we can expect a lot of instances.

**Comment by shine2lay** [comment]
i'll wrap the output in the calling function

**Comment by jusx** [comment]
Too many Instantiated objects are only a concern if you are running in
embedded systems with low memory availability. This is not the case for
mergeable. But if you insist it’ll be problematic I’m fine with other
methods to dry.

**Pull request merged**